src/include/ipc/topology.h

```diff
@@ -87,6 +87,12 @@ struct sof_ipc_comp {
 #define SOF_MEM_CAPS_EXEC		BIT(7) /**< executable */
 #define SOF_MEM_CAPS_L3			BIT(8) /**< L3 memory */
 
+#ifdef CONFIG_L3_HEAD
+#define SOF_MEM_CAPS_MAX_BIT		BIT(9) /**< Used for input validation */
+#else
+#define SOF_MEM_CAPS_MAX_BIT		BIT(8) /**< Used for input validation */
+#endif
```

Comment:
I would put this log in the FW src (based on Kconfig) as the ABI should not be subject to Kconfig.